### PR TITLE
Added more documentation

### PR DESCRIPTION
**Pull request by jaytaph**
Added additional documentation to structs and modules. Still a lot of room for improvement

### CODE REVIEW

**Review by blaumeise20 — CHANGES_REQUESTED** [review]
I have no idea if I'm "socially" allowed to review stuff in here, but I have one suggestion: don't use a "This blabla" style but a more present ... way (I forgot the name). So instead of writing
```
This module contains the HTML5 parser as specified by the W3C.
```
I would write
```
The HTML5 parser implemented according to the W3C specification.
```
Similar things apply for some other things like macros and structs. Essentially it should be consistent with the way methods/functions are documented, e.g.:
```
Returns the current offset in the stream
```

**Comment by jaytaph** [comment]
> I have no idea if I'm "socially" allowed to review stuff in here

everyone is :-)

I think it is a good idea. If you willing to help us out on this, that would be even better :)

**Comment by emwalker** [comment]
> I have no idea if I'm "socially" allowed to review stuff in here

That's how we all got here so far. :)

File: src/api.rs [hunk 1]
```diff
@@ -1,2 +1,8 @@
+//! This module contains all the APIs that are exposed through javascript
```
Comment (emwalker) [inline src/api.rs:1]:
```suggestion
//! All the APIs that are exposed through JavaScript
```

Just to run with @blaumeise20's thought.  He might have additional suggestions.

File: src/api/console.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//! This module contains the console api as described by <https://console.spec.whatwg.org/>
```
Comment (emwalker) [inline src/api/console.rs:1]:
```suggestion
//! The console API, as described by <https://console.spec.whatwg.org/>
```

File: src/html5_parser.rs [hunk 3]
```diff
@@ -1,3 +1,7 @@
+//! This module contains the HTML5 parser as specified by the W3C.
```
Comment (emwalker) [inline src/html5_parser.rs:1]:
```suggestion
//! The HTML5 parser, following the W3C specification
```

File: src/html5_parser/input_stream.rs [hunk 4]
```diff
@@ -41,29 +46,36 @@ impl fmt::Display for Position {
...
     Utf8(char),
-    /// Surrogate character (since they cannot be stored in <char>)
+    /// Surrogate character (since they cannot be stored in char)
     Surrogate(u16),
     /// End of stream
     Eof,
 }
 
 impl Element {
+    /// Returns true when the element is an EOF
```
Comment (emwalker) [inline src/html5_parser/input_stream.rs:63]:
```suggestion
    /// Whether the element is an EOF
```

You get the idea.  Similar suggestions apply further down.

**Review by Kiyoshika — APPROVED** [review]

**Comment by jaytaph** [comment]
Lets merge as-is, and change later when we have something more formalized

**Pull request merged**